src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/HandshakeResponseMessage.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.SignalR.Internal.Protocol
+{
+    public class HandshakeResponseMessage : HubMessage
+    {
+        private static readonly HandshakeResponseMessage _empty = new HandshakeResponseMessage(null);
```

Comment:
As with `CloseMessage`, could this just be a public field? That's what we've done in other situations like this.